pkg/controller/instance/controller_reconcile.go

```diff
@@ -361,64 +487,126 @@ func (igr *instanceGraphReconciler) handleInstanceDeletion(ctx context.Context)
 
 // initializeDeletionState prepares resources for deletion by checking their
 // current state and marking them appropriately.
-func (igr *instanceGraphReconciler) initializeDeletionState() error {
-	for _, resourceID := range igr.runtime.TopologicalOrder() {
-		if _, err := igr.runtime.Synchronize(); err != nil {
-			return fmt.Errorf("failed to synchronize during deletion state initialization: %w", err)
-		}
+func (igr *instanceGraphReconciler) initializeDeletionState(levels [][]string) error {
```

Comment:
Super high level but why do we even need initializeDeletionState still? I was assuming that this state should be covered by a proper pruned diff.